### ISSUE TITLE
Is it possible to configure cljfmt to support Tonsky's formatting suggestions?

### ISSUE THREAD

**Issue by PEZ** [weavejester/cljfmt#154]
I have been trying to figure out what `:indent` rules would make `cljfmt` to format according to [Tonsky's Better Clojure Formatting](http://tonsky.me/blog/clojurefmt/) suggestions. But all my attempts fail in different ways. Is it even possible, or does it require a change in the code?

**Comment by PEZ** [comment]
Specifically I would think `:indents ^:replace {#"^\w" [[:inner 0]]}` should work, but it seems to make all lists indent two spaces:
```clojure
(defn multi-arity
  ([x]
    body)
   ^----- Should indent here
  ([x y]
    body))
```
Which messes things up for Parinfer.

**Comment by weavejester** [comment]
That *should* work. I'm curious to know why it doesn't.

**Comment by PEZ** [comment]
Here is how I tested it:

```clojure
(comment
  (def t "(when something
  body)

(defn f [x]
  body)

(defn f
  [x]
  body)

(defn many-args [a b c
                 d e f]
  body)

(defn multi-arity
  ([x]
   body)
  ([x y]
   body))

(let [x 1
      y 2]
  body)

[1 2 3
 4 5 6]

{:key-1 v1
 :key-2 v2}

#{a b c
  d e f}

(or (condition-a)
  (condition-b))

(filter even?
  (range 1 10))

(clojure.core/filter even?
  (range 1 10))

(filter
 even?
  (range 1 10))")

  (def f
    (cljfmt/reformat-string t {:indents {#"^\w" [[:inner 0]]}}))

  (= t f)
  (println f))
```

(With or without the `^:replace` hint makes no difference for this test.)

**Comment by PEZ** [comment]
I started to suspect it could be a ClojureScript, or even a `shadow-cljs` thing, but it is the same in Clojure.

**Comment by ProjectFrank** [comment]
A new release was recently cut that incorporates a fix to inner indentation with an index: https://github.com/ProjectFrank/cljfmt/commit/f0ab087be527b94082e0ac335d19474e46860f73

Can you try again using `0.6.4`?

**Comment by ProjectFrank** [comment]
@PEZ any updates on whether this has been fixed?

**Comment by PEZ** [comment]
@ProjectFrank , I am sorry, I have forgotten to test this. I will merge it onto my fork and let you know what happens, this weekend.

**Comment by PEZ** [comment]
Yes! I'd say that this is now fixed. Many thanks!

Calva with tonsky-option is soon coming out... :smile:

**Comment by kawas44** [comment]
Hello,

Issue was closed but it does not look like version ```0.6.4``` fixes the multy-arity defn.
```diff
lein cljfmt check
Retrieving lein-cljfmt/lein-cljfmt/0.6.4/lein-cljfmt-0.6.4.pom from clojars
Retrieving cljfmt/cljfmt/0.6.4/cljfmt-0.6.4.pom from clojars
Retrieving lein-cljfmt/lein-cljfmt/0.6.4/lein-cljfmt-0.6.4.jar from clojars
Retrieving cljfmt/cljfmt/0.6.4/cljfmt-0.6.4.jar from clojars
src/play_with_fmt/core.clj has incorrect formatting
--- a/xxxxxxxxxxxxxxxxxxxxxxxxx/play-with-fmt/src/play_with_fmt/core.clj
+++ b/xxxxxxxxxxxxxxxxxxxxxxxx/play-with-fmt/src/play_with_fmt/core.clj
@@ -21,9 +21,9 @@

 (defn multi-arity
   ([x]
-   body)
+    body)
   ([x y]
-   body))
+    body))

1 file(s) formatted incorrectly
```
My lein cljfmt configuration
```clojure
  :cljfmt {:remove-consecutive-blank-lines? false
           :indents ^:replace {#"^[-*+!?_a-zA-Z]" [[:inner 0]]}}
```
